tests/chainer_tests/functions_tests/noise_tests/test_dropout.py

```diff
@@ -144,46 +148,49 @@ def test_immutable(self, backend_config):
     'specify_mask': [True, False],
     'train': [True, False],
 }))
+@testing.inject_backend_tests(
+    ['test_check'],
+    # CPU tests
+    testing.product({
+        'use_cuda': [False],
+        'use_ideep': ['never', 'always'],
+    })
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'use_cudnn': ['never', 'always'],
+    })
+)
 class TestDropoutMask(unittest.TestCase):
 
     def setUp(self):
         self.x = numpy.random.uniform(-1, 1, (2, 3)).astype(numpy.float32)
         self.mask = (numpy.random.uniform(-1, 1, (2, 3)) > 0).astype(
             numpy.float32)
 
-    def _check(self):
-        mask = self.mask if self.specify_mask else None
-        with chainer.using_config('train', self.train):
-            out, out_mask = functions.dropout(
-                self.x, 0.5, mask=mask, return_mask=True)
+    def test_check(self, backend_config):
+        mask = None
+        if self.specify_mask:
+            mask = self.mask
 
+        x, mask = backend_config.get_array((self.x.copy(), mask))
+        with chainer.using_config('train', self.train), backend_config:
+            out, out_mask = functions.dropout(
+                x, 0.5, mask=mask, return_mask=True)
         if self.train:
-            assert isinstance(out_mask, type(out.array))
-            if mask is None:
+            if mask is None and backend_config.use_cudnn != 'always':
+                assert isinstance(out_mask, type(out.array))
```

Comment:
Shouldn't this assertion always satisfied?

Reply:
Sorry, I mean it should be always checked.

Reply:
How can we get mask from cudnn implementation?

Reply:
```bash
Platform: Linux-4.9.0-8-amd64-x86_64-with-debian-9.8
Chainer: 7.0.0a1
NumPy: 1.16.2
CuPy:
  CuPy Version          : 7.0.0a1
  CUDA Root             : /usr/local/cuda
  CUDA Build Version    : 10000
  CUDA Driver Version   : 10000
  CUDA Runtime Version  : 10000
  cuDNN Build Version   : 7401
  cuDNN Version         : 7401
  NCCL Build Version    : None
  NCCL Runtime Version  : None
iDeep: Not Available

In [2]: x = cp.random.uniform(-1, 1, (10, 32)).astype(cp.float32)                                                                                                  
In [3]: with chainer.using_config('use_cudnn', 'always'): y, out_mask = F.dropout(x, return_mask=True)                                                             
In [4]: out_mask is None                                                                                                                                           
Out[4]: True
```

Reply:
How will I handle cudnn enabled forward?

Reply:
Hmm, how about enabling cuDNN only when `not return_mask`?
You will need to pass `return_mask` to the `Dropout` class.

Reply:
I think this is a preexisting bug.
Could you separate a PR, as we need to backport the fix?

Reply:
Or it's ok to just dodge the bug in this PR and fix it at some other time.

Reply:
OK, I'll send a PR to make `Dropout` FunctionNode consider `return_mask`, then I'll add tests that check cuDNN correctly used/unused.

Is my understanding correct?

Reply:
Yes. Thank you very much!